pkg/driver/controllerserver.go

```diff
@@ -239,9 +226,7 @@ func (cs *controllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVol
 	endPoint := secretMap["cosEndpoint"]
 	locationConstraint := secretMap["locationConstraint"]
 	sess := cs.cosSession.NewObjectStorageSession(endPoint, locationConstraint, creds, cs.Logger)
-	protectBucket := secretMap["bucketProtection"]
-
-	bucketToDelete, err := bucketToDelete(volumeID, protectBucket)
+	bucketToDelete, err := bucketToDelete(volumeID)
```

Comment:
The user provided bucket should not be deleted, even if `reclaimPolicy: Delete`
How we can protect a user provided bucket (pre-created by User)?

Reply:
New check added as discussed.